### PR TITLE
ci: add winget releaser workflow

### PR DESCRIPTION
**Pull request by sitiom**
[This action](https://github.com/vedantmgoyal9/winget-releaser) automatically generates manifests for Winget Community Repository ([microsoft/winget-pkgs](https://github.com/microsoft/winget-pkgs)) and submits them.

The Zen Browser package is being added to Winget (https://github.com/microsoft/winget-pkgs/pull/168940, https://github.com/microsoft/winget-pkgs/pull/168942), and this workflow will be used to update it.

**Before merging this:**
1. Ensure the `${{ secrets.DEPLOY_KEY }}` token has the `public_repo` permission:

example

2. Fork https://github.com/microsoft/winget-pkgs under @zen-browser. The action will use that fork for making a branch and creating a PR with the upstream [winget-pkgs](https://github.com/microsoft/winget-pkgs) repository on every release.
3. Install [Pull](https://github.com/apps/pull) on the winget-pkgs fork to ensure that it is constantly updated.

If you want to see an example of a PR created using this action, see [microsoft/winget-pkgs/pulls (Pull request has been created with WinGet Releaser)](https://github.com/microsoft/winget-pkgs/pulls?q=is%3Apr+sort%3Aupdated-desc+Pull+request+has+been+created+with+WinGet+Releaser).

### CODE REVIEW

File: .github/workflows/alpha.yml [hunk 1]
```diff
@@ -382,6 +382,31 @@ jobs:
...
+        with:
+          identifier: Zen-Team.Zen-Browser
+          version: ${{ needs.build-data.outputs.version }}
+          release-tag: ${{ needs.build-data.outputs.version }}
+          installers-regex: 'zen.installer-generic\.exe$'
+          token: ${{ secrets.DEPLOY_KEY }}
+      - name: Publish optimized build
+        uses: vedantmgoyal9/winget-releaser@main
+        with:
+          identifier: Zen-Team.Zen-Browser.Optimized
```
[the remark this answers is not in the document]
Comment (sitiom) [reply .github/workflows/alpha.yml:404]: See https://github.com/microsoft/winget-pkgs/pull/168940#issuecomment-2308625683

**Comment by sitiom** [comment]
> > 1. Ensure the ${{ secrets.DEPLOY_KEY }} token has the public_repo permission
> 
> Could we get that done in the CI by using the PAT from GH actions and [specifying it's permissions](https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#permissions)?

No, you can't. The PAT is used to make a commit and create a PR from an external repo.

> Sidenote, [ZenBrowser.Zen](https://github.com/microsoft/winget-pkgs/pull/168894) also exists, I don't know if we can do something about there now being 2 IDs for the browser.

- https://github.com/microsoft/winget-pkgs/pull/169157

File: .github/workflows/alpha.yml [hunk 2]
```diff
@@ -382,6 +382,31 @@ jobs:
             zen.macos-x64.dmg
             zen.macos-aarch64.dmg
 
+  release-winget:
+    if: ${{ github.event.inputs.create_release == 'true' }}
+    permissions: write-all
+    name: Release Winget
+    needs: [release, build-data]
```
[the remark this answers is not in the document]
Comment (sitiom) [reply .github/workflows/alpha.yml:389]: No, `release` is already dependent on the windows job, and `release-winget` does not need any artifact from the windows job (unlike `release-flatpak`)

**Review by sitiom — COMMENTED** [review]
Issue: https://github.com/zen-browser/desktop/issues/37#issuecomment-2315256821:

> > > In a couple of days, I'll receive a mail with a physical key and I'll need to figure out how to use it
> > >
> > @mauro-balades How can you automate signing builds with a physical key?
>
> I can't, I'll have to sign it and reupload

The workflow will pick up on the "unsigned" binary before you reupload it. Physical keys are incompatible with CI's. Perhaps you should have gone with the "Cloud" solution with Certum or [Azure Trusted Signing](https://azure.microsoft.com/en-us/products/trusted-signing) from Microsoft.

As a workaround, maybe the workflow should create a draft release first and then publish it altogether once you've signed the binaries?

**Comment by Zykkl** [comment]
Would using `on: release` avoid/fix the issue? So rather than it running with the main pipeline, it would run by itself whenever a new release is out. (Example https://github.com/microsoft/devhome/blob/94fde2349d7666575f28daf54451668f0acdbf4c/.github/workflows/winget-submission.yml#L3C1-L6C23).

**Comment by mauro-balades** [comment]
How would we sign it though? My license key is physical

**Comment by Zykkl** [comment]
The action would only run when you publish a release, so you'd still be able to sign your stuff

**Comment by mauro-balades** [comment]
I just want to get everything signed and working before expanding to more windows sites... But this will be merged

**Comment by sitiom** [comment]
Closed?

**Comment by UnownPlain** [comment]
Try changing the base branch to `dev` or opening a new PR

**Pull request closed without merging**